main.js

```diff
@@ -5,54 +5,77 @@ const { spawn } = require('child_process')
 
 const appPath = app.getAppPath()
 const ignoredPaths = /node_modules|[/\\]\./
-// Main file poses a special case, as its changes are
-// only effective when the process is restarted (hard reset)
-// We assume that electron-reload is required by the main
-// file of the electron application
-const mainFile = module.parent.filename
 
 /**
  * Creates a callback for hard resets.
  *
- * @param {String} eXecutable path to electron executable
- * @param {String} hardResetMethod method to restart electron
+ * @param {string} eXecutable path to electron executable
+ * @param {string} hardResetMethod method to restart electron
+ * @param {string[]} hardResetMethod method to restart electron
  * @returns {Function} handler to pass to chokidar
  */
-const createHardresetHandler = (eXecutable, hardResetMethod, argv) =>
-  () => {
-    // Detaching child is useful when in Windows to let child
-    // live after the parent is killed
-    const args = (argv || []).concat([appPath])
-    const child = spawn(eXecutable, args, {
-      detached: true,
-      stdio: 'inherit'
-    })
-    child.unref()
-    // Kamikaze!
+const createHardresetHandler = (eXecutable, hardResetMethod, argv) => () => {
+  // Detaching child is useful when in Windows to let child
+  // live after the parent is killed
+  const args = [appPath].concat(argv || [])
+  const child = spawn(eXecutable, args, {
+    detached: true,
+    stdio: 'inherit'
+  })
+  child.unref()
+  // Kamikaze!
 
-    // In cases where an app overrides the default closing or quiting actions
-    // firing an `app.quit()` may not actually quit the app. In these cases
-    // you can use `app.exit()` to gracefully close the app.
-    if (hardResetMethod === 'exit') {
-      app.exit()
-    } else {
-      app.quit()
-    }
+  // In cases where an app overrides the default closing or quiting actions
+  // firing an `app.quit()` may not actually quit the app. In these cases
+  // you can use `app.exit()` to gracefully close the app.
+  if (hardResetMethod === 'exit') {
+    app.exit()
+  } else {
+    app.quit()
   }
+}
 
+/**
+ * @typedef {Object} ExtraOptions
+ * @property {string} [hardResetMethod]
+ * @property {string[]} [argv]
+ * @property {boolean} [forceHardReset]
+ * @property {string} [electron]
+ * @property {string} [mainFile]
+ */
+
+/**
+ * @param {string | string[]} glob
+ * @param { Partial<chokidar.WatchOptions & ExtraOptions> } [options]
+ * @returns {void}
+ */
 module.exports = (glob, options = {}) => {
+  // Main file poses a special case, as its changes are
+  // only effective when the process is restarted (hard reset)
+  // We assume that electron-reload is required by the main
+  // file of the electron application if a path is not provided
+  const mainFile =
+    options.mainFile || module.parent ? module.parent.filename || '' : ''
```

Comment:
Adding explicit option values would simplify this (see comment above). I also don't think that an empty string is the best option here.

Btw. why do you check if `module.parent` is undefined/empty?

Reply:
Removed extra falsey check and defaulted mainFile option